### PR TITLE
chore: fixes for `simp` refactor

### PR DESCRIPTION
**Pull request by leodemoura**
This PR fixes issues created by https://github.com/leanprover/lean4/pull/3210

### CODE REVIEW

**Comment by digama0** [comment]
@semorrison I think this needs a different base branch, and/or a lean-pr-releases branch

**Comment by semorrison** [comment]
I've changed the base to `bump/v4.6.0`, and also merged that branch into this PR to minimise the displayed diffs.

**Review by semorrison — APPROVED** [review]
LGTM. Once leanprover/lean4#3210 lands in a nightly, we can update this PR and then merge into `bump/v4.6.0`.

**Review by digama0 — APPROVED** [review]

File: Std/Tactic/NormCast/Ext.lean [hunk 1]
```diff
@@ -5,6 +5,7 @@ Authors: Paul-Nicolas Madelaine, Robert Y. Lewis, Mario Carneiro, Gabriel Ebner
 -/
 import Lean.Meta.CongrTheorems
 import Lean.Meta.Tactic.Simp.SimpTheorems
+import Lean.Meta.Tactic.Simp.Simproc
```
Comment (digama0) [inline Std/Tactic/NormCast/Ext.lean:8]:
```suggestion
```

**Pull request merged**